### PR TITLE
SNOW-895534: Add HTAP query context struct

### PR DESCRIPTION
**Pull request by sfc-gh-pfus**
### Description
Provided query context structs and tests for serialisation and deserialisation.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: htap.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package gosnowflake
+
+type queryContextEntry struct {
```
Comment (sfc-gh-dprzybysz) [inline htap.go:3]: you are adding one struct but PR says that there will be struct**s** - are you going to add more of them in this PR?

File: htap.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package gosnowflake
```
Comment (sfc-gh-dprzybysz) [inline htap.go:1]: you don't create separate packages for the features?

File: htap_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package gosnowflake
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestDeserializeOpaqueContext(t *testing.T) {
```
Comment (sfc-gh-dprzybysz) [inline htap_test.go:10]: you test deserialization and serialization in the test? maybe it should reflect it in the test name

also consider using Go specific names - decode and marshal?

File: htap_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			err := json.NewDecoder(strings.NewReader(tc.json)).Decode(&qc)
+			if err != nil {
+				t.Fatalf("failed to decode json. %v", err)
+			}
+
+			if !reflect.DeepEqual(tc.qc, qc) {
+				t.Errorf("failed to decode json. expected: %v, got: %v", tc.qc, qc)
+			}
+
+			bytes, err := json.Marshal(tc.qc)
```
Comment (sfc-gh-dprzybysz) [inline htap_test.go:82]: let's marshal `qc`, not expected structure

File: htap.go (hunk 1 above)
Comment (sfc-gh-pfus) [reply htap.go:3]: No, that's it. Changed commit msg.

File: htap.go (hunk 2 above)
Comment (sfc-gh-pfus) [reply htap.go:1]: No, everything is in the `gosnowflake` package.

File: htap_test.go (hunk 3 above)
Comment (sfc-gh-pfus) [reply htap_test.go]: Done.

File: htap_test.go (hunk 4 above)
Comment (sfc-gh-pfus) [reply htap_test.go]: Done.

**Review by sfc-gh-pbulawa — APPROVED** [review]

File: htap.go (hunk 1 above)
Comment (sfc-gh-dprzybysz) [reply htap.go:3]: Let's change also the PR name

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Pull request merged**